tests/dedicatedvstests/l7_dedicated_graph_test.go

```diff
@@ -182,6 +182,41 @@ func SetupDomain() {
 	mcache.CloudKeyCache.AviCacheAdd("Default-Cloud", cloudObj)
 }
 
+func SetUpIngressForCacheSyncCheckMultiPaths(t *testing.T, tlsIngress, withSecret bool, fqdns []string, paths []string, modelNames ...string) {
+	SetupDomain()
+	SetUpTestForIngress(t, modelNames...)
+	ingressObject := integrationtest.FakeIngress{
+		Name:        "app-root-test",
+		Namespace:   "default",
+		DnsNames:    fqdns,
+		Ips:         []string{"8.8.8.8"},
+		HostNames:   []string{"v1"},
+		Paths:       paths,
+		ServiceName: "avisvc",
+	}
+	if withSecret {
+		integrationtest.AddSecret("my-secret", "default", "tlsCert", "tlsKey")
+	}
+	if tlsIngress {
+		ingressObject.TlsSecretDNS = map[string][]string{
+			"my-secret": {"foo.com"},
+		}
+	}
+	ingrFake := ingressObject.Ingress()
+	if _, err := KubeClient.NetworkingV1().Ingresses("default").Create(context.TODO(), ingrFake, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("error in adding Ingress: %v", err)
+	}
+	integrationtest.PollForCompletion(t, modelNames[0], 5)
+}
+
+func TearDownIngressForCacheSyncCheckPath(t *testing.T, modelName string) {
```

Comment:
These functions have been defined for each package in _test.go files and using global variables shared with other functions in the same file. Also, variables defined in _test.go files are not accessible from normal .go files so adding these in a lib.go will also cause complexities. If required, this will have to be taken up separately.